components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,522 @@
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Class to call BPMN backend
+ */
+public class BPMNExplorerServiceImpl implements BPMNExplorerService {
+    private static final Logger log = LoggerFactory.getLogger(BPMNExplorerServiceImpl.class);
+
+    private CloseableHttpClient client;
+    private String baseUrl;
+    private String userName;
+    private String passWord;
+
+    /**
+     * Initialize the HttpClient
+     *
+     * @param host
+     * @param port
+     */
+    private void setHttpClient(String host, int port) {
+        try {
+            client = HttpClients.custom()
+                    .setSSLSocketFactory(new SSLConnectionSocketFactory(
+                            SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build()))
+                    .build();
+        } catch (NoSuchAlgorithmException | eyManagementException | KeyStoreException e) {
+            log.error("Exception : ", e);
+        }
+        baseUrl = "https://" + host + ":" + port;
+        //Need to change this part after introducing user managemnet module
+        userName = "admin";
+        passWord = "admin";
+    }
+
+    /**
+     * Return BPMN process details from deployments
+     *
+     * @param pagination
+     * @param host
+     * @param port
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public JSONArray getProcessDetails(String pagination, String host, int port, int tenantId) throws IOException {
+
+        HttpGet httpGet = null;
+        CloseableHttpResponse response = null;
+        BufferedReader rd = null;
+        String line;
+        JSONArray returnDataArray = new JSONArray();
+        String cookie;
+        int start = 0;
+
+        setHttpClient(host, port);
+
+        if (pagination.equals("")) {
+            start = 0;
+        } else {
+            int pageCount = Integer.parseInt(pagination.split("=")[1]);
+            start = (pageCount - 1) * 10;
+        }
+
+
+        try {
+            cookie = HTTPAuthenticate();
+            httpGet = new HttpGet(
+                    new URI(baseUrl + "/bpmn/repository/deployments?tenantId=" + tenantId + "&start=" + start));
+            httpGet.addHeader("Content-type", "application/json");
+            String authString = userName + ":" + password;
+            byte[] authEncBytes = Base64.encodeBase64(authString.getBytes());
+            httpGet.addHeader("Authorization", "Basic" + new String(authEncBytes));
+
+            response = client.execute(httpGet);
+            rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent(), Charset.defaultCharset()));
+            StringBuffer resultList = new StringBuffer();
+            while ((line = rd.readLine()) != null) {
+                resultList.append(line);
+            }
```

Comment:
Can you try with resources option for cleaning up resources here. 
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html